cmd/dstask.go

```diff
@@ -63,6 +64,21 @@ func main() {
 			ts.SavePendingChanges()
 			dstask.MustGitCommit("Added %s", task)
 		}
+	case dstask.CMD_RM, dstask.CMD_REMOVE:
+		if len(cmdLine.IDs) < 1 {
+			dstask.ExitFail("%s", "missing argument: id")
+		}
+		ts := dstask.LoadTasksFromDisk(dstask.NON_RESOLVED_STATUSES)
+		for _, id := range cmdLine.IDs {
+			task := ts.MustGetByID(id)
+			relativePath := filepath.Join(task.Status, task.UUID+".yml")
+			dstask.MustGitRemove(relativePath)
+			// NOTE: do we need to do anything with the TaskSet tasksByUUID map?
+			// *ts.tasksByUUID[task.UUID]
+			// NOTE: Do we need to interact with the TaskSet or the IdsMap database?
```

Comment:
Technically this won't matter as the IDs are recomputed every time `SavePendingChanges()` is called.

However I think it is cleaner to keep all task set management within the taskset methods. Perhaps each task could have a boolean `delete` which is never serialised -- `task.save()` would then perform the deletion when called by `SavePendingChanges()`. The interface to this could be via `MustUpdateTask()` given a mutated task returned by your existing call to `MustGetByID`

Reply:
I'll rework this to encapsulate the functionality within the taskset. :+1: